server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/RareTermsAggregatorFactory.java

```diff
@@ -44,6 +47,89 @@
+     */
+    private static RareTermsAggregatorSupplier bytesSupplier() {
+        return new RareTermsAggregatorSupplier() {
+            @Override
+            public Aggregator build(String name,
+                                    AggregatorFactories factories,
+                                    ValuesSource valuesSource,
+                                    DocValueFormat format,
+                                    int maxDocCount,
+                                    double precision,
+                                    IncludeExclude includeExclude,
+                                    SearchContext context,
+                                    Aggregator parent,
+                                    List<PipelineAggregator> pipelineAggregators,
+                                    Map<String, Object> metaData) throws IOException {
+
+                ExecutionMode execution = ExecutionMode.MAP; //TODO global ords not implemented yet, only supports "map"
+
+                if ((includeExclude != null) && (includeExclude.isRegexBased()) && format != DocValueFormat.RAW) {
+                    throw new AggregationExecutionException("Aggregation [" + name + "] cannot support " +
+                        "regular expression style include/exclude settings as they can only be applied to string fields. " +
+                        "Use an array of values for include/exclude clauses");
+                }
+
+                return execution.create(name, factories, valuesSource, format,
+                    includeExclude, context, parent, pipelineAggregators, metaData, maxDocCount, precision);
+
+            }
+        };
+    }
+
+    /**
+     * This supplier is used for all fields that expect to be aggregated as a numeric value.
+     * This includes floating points, and formatted types that use numerics internally for storage (date, boolean, etc)
+     */
+    private static RareTermsAggregatorSupplier numericSupplier() {
+        return new RareTermsAggregatorSupplier() {
+            @Override
+            public Aggregator build(String name,
+                                    AggregatorFactories factories,
+                                    ValuesSource valuesSource,
+                                    DocValueFormat format,
+                                    int maxDocCount,
+                                    double precision,
+                                    IncludeExclude includeExclude,
+                                    SearchContext context,
+                                    Aggregator parent,
+                                    List<PipelineAggregator> pipelineAggregators,
+                                    Map<String, Object> metaData) throws IOException {
+
+                if ((includeExclude != null) && (includeExclude.isRegexBased())) {
+                    throw new AggregationExecutionException("Aggregation [" + name + "] cannot support regular expression " +
```

Comment:
It is somewhat conditional, since the same include/exclude could work on a different field type (strings vs numerics).  But I generally agree, will switch it over.